### PR TITLE
Terraform `merge` function handling

### PR DESCRIPTION
**Pull request by robeden**
This also does a better job at pulling out "eval" blocks for processing.

Fixes: #764

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by schosterbarak** [comment]
@nimrodkor @rotemavni can you take a look?

File: tests/terraform/parser/resources/parser_scenarios/merge_function/main.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    Tag2 = "two"
+  }
+  common_data_tags = {
+    Tag3 = "three"
+  }
+
+  local_to_local = merge(local.common_tags, local.common_data_tags)
+  local_to_manual = merge(local.common_tags, {Tag4 = "four"})
+  manual_to_local = merge({Tag4 = "four"}, local.common_tags)
+  manual_to_manual = merge({Tag4 = "four"}, {Tag5 = "five"})
```
Comment (nimrodkor) [inline tests/terraform/parser/resources/parser_scenarios/merge_function/main.tf:13]: @robeden I've seen many cases where the merge is not a oneliner, i.e.:
```terraform
local_to_manual = merge(
    local.common_tags,
    {Tag4 = "four"},
    {Tag2="Dev"}
)
```

Can you add a test case to ensure this is working properly?

**Review by nimrodkor — COMMENTED** [review]
@robeden Very nice! This is great!
Just one edge case I would love to make sure that works.

File: tests/terraform/parser/resources/parser_scenarios/merge_function/main.tf (hunk 1 above)
Comment (robeden) [reply tests/terraform/parser/resources/parser_scenarios/merge_function/main.tf]: Good call because the HCL2 parser returns some [weird results](https://github.com/bridgecrewio/checkov/pull/771/commits/44b5dd3e8119b3b2e8340acbff0e5b964b457d53#diff-23faf3382230bc5b99ec03b5e221636a628de63406746fe5063c5c5aa80ec11fR40) when that happens. It handles it now.

**Review by nimrodkor — APPROVED** [review]

**Comment by robeden** [comment]
@rotemavni @schosterbarak - Any chance we could move this through? 🥺

**Comment by schosterbarak** [comment]
@nimrodkor has done his charm. Thanks mate!  👍

**Comment by robeden** [comment]
Thanks!
